compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2381,7 +2391,8 @@ where
 
                     match tcx.struct_tail_erasing_lifetimes(pointee, cx.param_env()).kind() {
                         ty::Slice(_) | ty::Str => TyMaybeWithLayout::Ty(tcx.types.usize),
-                        ty::Dynamic(_, _) => {
+                        // FIXME(eholk): Do the right thing with trait object representation
+                        ty::Dynamic(_, _, _repr) => {
```

Comment:
This should just be `ty::Dyn`, `&dyn* Trait` doesn't have metadata since it's sizd